core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2798,7 +2798,7 @@ object SparkContext extends Logging {
         defaultProf.maxTasksPerExecutor(sc.conf) < cpuSlots) {
         throw new IllegalArgumentException("The number of slots on an executor has to be " +
           "limited by the number of cores, otherwise you waste resources and " +
-          "dynamic allocation doesn't work properly. Your configuration has " +
+          "some scheduling doesn't work properly. Your configuration has " +
```

Comment:
We should also update the comment above

Reply:
Shall we give one or two examples for the "some scheduling" here?

Reply:
I don't see the point of giving  the examples, this is a temporary error that will go away when when the rest of the feature is put in.  The end user just needs to know cores has to be limiting factor. All they need to know is spark won't work right if you don't.

Reply:
The comment above already says "dynamic allocation and scheduler ".  I think that covers most of the cases.